variable/creation.cpp

```diff
@@ -36,6 +36,8 @@ Variable empty_like(const Variable &prototype,
 
 Variable special_like(const Variable &prototype, const FillValue &fill) {
   const char *name = "special_like";
+  if (fill == FillValue::Default)
```

Comment:
You can use a switch instead. Then clang will warn if an enum field is not handled.

Reply:
I don't think that is true for all compilers? Or is it in a recent version of the standard? Also, what if someone cast an int to the enum?

Reply:
No, AFAIK only clang does it. But I occasionally build with clang and fix its warnings, so this is better than nothing.

> Also, what if someone cast an int to the enum?

That only makes sense if they cast to one of the defined fields, so this would still be covered.

Reply:
Well, but what if they don't? Returning an uninitialized value?

Reply:
What returns an uninitialised value? It doesn't matter if you use a switch of if-else, if you forget a case, bad things happen.

Reply:
Ok, so you would still leave the `throw`?

Reply:
Yes, use a `switch` and let it fall through to `throw` if no case matches.

Reply:
But don't use `default`. That disables the compiler check.